heroku-22/installed-packages.txt

```diff
@@ -306,7 +306,6 @@ mawk
 media-types
 mlock
 mount
-mtools
```

Comment:
Removing packages is a breaking change, and since Heroku apps are continuously rebased on newer stack images (so that they pick up security updates), any changes like that will not only break new builds moving forwards, but existing apps too (even those that haven't been built recently).

As such, this removal (and the syslinux change below) are not something we can accept for existing stacks.

Options are to:
1. Try and find a way to still include this package on all architectures
2. (If the package just doesn't exist for ARM64) Keep the package for x86_64, and omit on ARM64 only. (Though this then means there's even less parity between images, meaning yet another dev-prod parity difference.)
3. Wait until Heroku-24, where the package can be removed from all architectures

Reply:
This package seems to be available on arm64 on all recent Ubuntu versions, no?

https://packages.ubuntu.com/bionic/mtools

Reply:
mtools is not explicitly included, but is a dependency of syslinux. It can be required explicitly for ARM64, yes. Syslinux however cannot. I did some research and it seems that package is almost specifically built for x86, so there is no way to include that package on all architectures. 

So, to make this work we have to go with either option 2 or 3 as @edmorley suggested. However, Heroku 22 is only just out, so waiting for Heroku 24 could be a while.

@edmorley do we actually need a modified installed-packages.txt for ARM64 tho? Where is that used, in Heroku itself only right? And Heroku doesn't use ARM64 servers right? My original idea was that if you'd build this Docker image in the original way, it would just build the x86 image the way it was before. Only when you'd use buildx it would also build for arm64, which could differ a bit, yes.

Reply:
> do we actually need a modified installed-packages.txt for ARM64 tho? Where is that used, in Heroku itself only right?

The file is effectively a test fixture, it's not used outside of this repository. It's there to avoid hidden package changes as upstream make changes, to avoid regressions like #71. (Some packages will occasionally change their transitive dependency tree, which is not a breaking change for that particular package, but is a breaking change for the stack-images as a whole, if that only-installed-transitively package was something end users were depending on.)

As such, yes we need this test for all architectures.

Reply:
Is there an easy way to make APT skip a package without failing? So we can just require syslinux but it will omit it for ARM64? Otherwise we would have to create something like a multi-stage docker build perhaps?

Also, `write_package_list` can be done for different architectures by referencing the SHA265 hash - see [Examples](https://docs.docker.com/build/building/multi-platform/#example) here

Reply:
A multi-stage build is not what we need here. Bear in mind that the x86_64 and arm64 Docker images will be built separately, whereas multi-stage builds refer to a single build that is then performed in several stages.

The solution needed is:

> Doing this would require adding architecture conditionals to build.sh

ie: Adding bash conditionals that check the output of `uname -m`.

Reply:
This is actually possible with buildx:

```bash
if [ "$TARGETARCH" == "amd64" ]; then apt-get install -y --no-install-recommends syslinux; fi
```

Reply:
I have updated the PR with this logic